### PR TITLE
Test schema compatibility in typedBigQuery

### PR DESCRIPTION
**Pull request by ravwojdyla**

### CODE REVIEW

**Comment by ravwojdyla** [comment]
A couple of changes here:
 * 7d2655c - `bigquery.project` if set, trumps Dataflow's `GcpOptions` - which I think makes sense
 * 7d2655c - validate that `GcpOptions.getProject` is not null, before we pass it to BigQuery client
 * d068055 - initial tests for schema compatibility in `typedBigQuery` and integration tests for Scio pipeline with BigQuery
 * f4c0a20 - BigQuery scala files are not needed on travis

**Comment by ravwojdyla** [comment]
@nevillelyh @andrewsmartin review pls?

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala [hunk 1]
```diff
@@ -233,4 +244,10 @@ package object experimental {
 
   }
 
+  private def areSchemasCompatible(from: TableSchema, to: TableSchema): Boolean = {
+    import scala.collection.JavaConverters._
+    val fields2 = to.getFields.asScala
+    from.getFields.asScala.forall(fields2.contains(_))
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: What about nested RECORDs?

**Comment by andrewsmartin** [comment]
👍 LGTM

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala [hunk 2]
```diff
@@ -85,8 +86,18 @@ package object experimental {
...
         val table = scala.util.Try(BigQueryIO.parseTableSpec(newSource)).toOption
         if (table.isDefined) {
+          val newTableSchema = bigQueryClient.getTableSchema(table.get)
+          require(areSchemasCompatible(bqt.schema, newTableSchema),
+            s"""|New source $newSource has incompatible schema.
+                |New source schema ${newTableSchema.toString},
+                |Original schema ${bqt.schema.toString}!""".stripMargin)
           self.bigQueryTable(table.get)
         } else {
+          val newQuerySchema = bigQueryClient.getQuerySchema(newSource)
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: This `newQuerySchema` may still be from view if query is legacy syntax. How do you plan to fix this?

**Comment by ravwojdyla** [comment]
@nevillelyh could you give reproducible example that current logic breaks.

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala (hunk 1 above)
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: Add some extra tests.

File: scio-core/src/main/scala/com/spotify/scio/experimental/package.scala (hunk 2 above)
Comment (ravwojdyla) [inline scio-core/src/main/scala/com/spotify/scio/experimental/package.scala]: Out of scope for this change.

**Comment by ravwojdyla** [comment]
There must be something with the integration tests and this PR, cause other PR are relatively fine +/- the unpredictable sampling algorithms.

**Comment by nevillelyh** [comment]
This has been sitting for a long time. IMHO I prefer to abandon this since it's not critical and diverges further from upstream BigQuery handling mechanism.

**Comment by ravwojdyla** [comment]
There is nothing wrong with the logic here - only that travis is timing out. I would like to find out why, cause this PR also introduces some IT tests.

**Pull request closed without merging**